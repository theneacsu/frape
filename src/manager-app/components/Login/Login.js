import React from 'react'
import { connect } from 'react-redux'
import { startLogin } from '../../../actions/auth/auth'
import styles from './Login.module.css'

export const Login = props => (
  <div className={styles.wrapperDiv}>
    <div className={styles.contentDiv}>
      <h1 className={styles.h1}>Holistic Integration</h1>
      <h2 className={styles.h2}>
        Increase your performance, focus and productivity
      </h2>
      <button className={styles.button} onClick={props.startLogin}>
        Get started with Google
      </button>
    </div>
  </div>
)

const mapDispatchToProps = dispatch => ({
  startLogin: () => dispatch(startLogin())
})

export default connect(
  undefined,
  mapDispatchToProps
)(Login)
