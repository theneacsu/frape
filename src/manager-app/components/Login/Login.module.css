.wrapperDiv {
  background: url('/images/main.jpg');
  background-size: cover;
  display: flex;
  justify-content: center;
  height: 100vh;
  width: 100vw;
}

.contentDiv {
  text-align: center;
  padding: 3rem;
  margin: 7rem 5rem 0;
  letter-spacing: 0.1rem;
  width: 320px;
}

.h1 {
  margin: 0 0 6rem 0;
  line-height: 1;
  letter-spacing: 0.2rem;
}

.h2 {
  margin: 0 0 5rem 0;
}

.button {
  background: #dd4b39;
  color: white;
  border: none;
  padding: 1.6rem;
  border-radius: 5rem;
  font-size: 1.6rem;
  outline: none;
}

.button:hover {
  cursor: pointer;
}
