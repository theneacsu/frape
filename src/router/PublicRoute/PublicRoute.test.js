import React from 'react'
import { shallow } from 'enzyme'
import { PublicRoute } from './PublicRoute'
import Login from '../../manager-app/components/Login/Login'

test('Should redirect to Dashboard when the user is authenticated', () => {
  const wrapper = shallow(
    <PublicRoute path="/dashboard" component={Login} isAuthenticated={true} />
  )
  expect(wrapper).toMatchSnapshot()
})

test('Should render correcyly the Login component when the user is not authenticated', () => {
  const wrapper = shallow(
    <PublicRoute path="/dashboard" component={Login} isAuthenticated={false} />
  )
  expect(wrapper).toMatchSnapshot()
})
