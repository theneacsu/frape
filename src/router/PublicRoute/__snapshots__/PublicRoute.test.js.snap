// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should redirect to Dashboard when the user is authenticated 1`] = `
<Route
  component={[Function]}
  path="/dashboard"
/>
`;

exports[`Should render correcyly the Login component when the user is not authenticated 1`] = `
<Route
  component={[Function]}
  path="/dashboard"
/>
`;
